src/psyclone/tests/generator_test.py

```diff
@@ -745,6 +745,46 @@ def test_main_expected_fatal_error(capsys):
     assert output == expected_output
 
 
+def test_generate_trans_error(tmpdir, capsys, monkeypatch):
+    '''Test that a TransformationError exception in the generate function
+    is caught and output as expected by the main function.  The
+    exception is only raised with the new PSyIR approach to modify the
+    algorithm layer which is currently in development so is protected
+    by a switch. This switch is turned on in this test by
+    monkeypatching.
+
+    '''
+    monkeypatch.setattr(generator, "LFRIC_TESTING", True)
+    code = (
+        "module setval_c_mod\n"
+        "contains\n"
+        "subroutine setval_c()\n"
+        "  use builtins\n"
+        "  use constants_mod, only: r_def\n"
+        "  use field_mod, only : field_type\n"
+        "  type(field_type) :: field\n"
+        "  real(kind=r_def) :: value\n"
+        "  call invoke(setval_c(field, value))\n"
+        "end subroutine setval_c\n"
+        "end module setval_c_mod\n")
+    filename = str(tmpdir.join("alg.f90"))
+    with open(filename, "w", encoding='utf-8') as my_file:
+        my_file.write(code)
+    with pytest.raises(SystemExit) as excinfo:
+        main([filename])
+    # the error code should be 1
+    assert str(excinfo.value) == "1"
+    _, output = capsys.readouterr()
+    expected_output = (
+        "Transformation Error: Error in RaisePSyIR2LFRicAlgTrans "
```

Comment:
As I said earlier, I'm not convinced that a call to 'generate' should raise a TransformationError but we can discuss. Perhaps we need a new `RaisingError` or something?

Reply:
Modified as agreed.